devicemodel/hw/pci/virtio/virtio_ipu.c

```diff
@@ -0,0 +1,395 @@
+/*
+ * Copyright (C) 2018 Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ */
+
+/*
+ * virtio ipu
+ * IPU mediator DM
+ */
+
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <assert.h>
+#include <pthread.h>
+
+#include "dm.h"
+#include "pci_core.h"
+#include "virtio.h"
+#include "virtio_kernel.h"
+#include "vmmapi.h"
+
+/*
+ * Size of queue was chosen experimentally in a way
+ * that it allows to do IPU ISYS MIPI camera capture without
+ * any delay for interrupt/msg send, this value should
+ * be tuned.
+ */
+#define VIRTIO_IPU_RINGSZ 1024
+
+/*
+ * IPU mediation makes use of 2 VQs.
+ * VQ0 for buffer mgmnt & 1 time configuration;
+ * VQ1 for interrupt/msg exchange
+ */
+#define VIRTIO_IPU_VQ_NUM 2
+
+const char *ipu4_vbs_dev_path = "/dev/vbs_ipu";
+
+static int virtio_ipu_debug = 0;
+#define DPRINTF(...)\
+do {\
+	if (virtio_ipu_debug)\
+		printf(__VA_ARGS__);\
+} while (0)
+
+#define WPRINTF(...) printf(__VA_ARGS__)
```

Comment:
I see all the DPRINTF/WPRINTF in this file are include the virtio_ipu at the head.
Can we update the DPRINTF/WPRINTF with virtio_ipu prefix? You can refer to the UPRINTF defined in include/usb_core.h

Reply:
Replaced with UPRINTF as recommended. The log level flag is not defined as macro because 8.1 gcc is throwing following note message at time of compilation

hw/pci/virtio/virtio_ipu.c:50:18: note: in expansion of macro ‘IPU_LOG_LEVEL’
  do { if (lvl <= IPU_LOG_LEVEL) printf(LOG_TAG fmt, ##args); } while (0)

Reply:
@kushal179 The debug level is fine. You don't have to support it if you needn't. Just suggest that to add prefix by default to avoid every PRINTF include them.

Reply:
Sure. Please let me know if there are any more feedback. Thank you